bokehjs/src/lib/models/glyphs/image.ts

```diff
@@ -48,6 +52,55 @@ export class ImageView extends XYGlyphView {
     }
   }
 
+  _index_data(): SpatialIndex {
+    const points = [];
+    for (let i = 0, end = this._x.length; i < end; i++) {
+      const [l, r, t, b] = this._lrtb(i);
+      if (isNaN(l+r+t+b) || !isFinite(l+r+t+b)) {
+        continue;
+      }
+      points.push({minX: l, minY: b, maxX: r, maxY: t, i});
+    }
+    return new SpatialIndex(points);
+  }
+
+  _lrtb(i: any) : any{
```

Comment:
``i: number`` is not sufficient here, certainly looks like it should be based on the way it is used.